src/main/java/frc/robot/SuperStructure.java

```diff
@@ -86,6 +90,35 @@ public SuperStructure(Elevator elevator, Arm arm) {
         SmartDashboard.putData("Superstructure", mechanism);
     }
 
+    private double getMinimumAllowableElevatorPosition(double shoulderAngle, double wristAngle) {
+        return Math.max(
+                elevator.getConstants().minimumPosition(),
+                Math.max(
+                        -(arm.getConstants().shoulderLength() + SHOULDER_BUFFER) * Math.sin(shoulderAngle),
+                        arm.getConstants().shoulderLength() * Math.sin(shoulderAngle) - (arm.getConstants().wristLength() + WRIST_BUFFER)
+                                * Math.sin(shoulderAngle + wristAngle)));
+    }
+
+    private double getMinimumAllowableWristAngle(double elevatorPosition, double shoulderAngle) {
+        double length = (arm.getConstants().shoulderLength() * Math.sin(shoulderAngle) + elevatorPosition - WRIST_BUFFER) / arm.getConstants().wristLength();
+        if (Math.abs(length) <= 1.0) {
+            return Math.max(arm.getConstants().minimumWristAngle(), -shoulderAngle - Math.asin(length));
+        }
+        return arm.getConstants().minimumWristAngle();
+    }
+
+    private double getMinimumAllowableShoulderAngle(double currentElevatorPos, double currentShoulderPos, double currentWristPos,
+                                                    double targetElevatorPos, double targetShoulderPos, double targetWristPos) {
+        double targetPos = targetShoulderPos;
+        if (((currentElevatorPos - SHOULDER_BUFFER) / arm.getConstants().shoulderLength()) <= 1) {
+            targetPos = Math.max(targetPos, -Math.asin((currentElevatorPos - SHOULDER_BUFFER) / arm.getConstants().shoulderLength()));
+        }
+        if (((targetElevatorPos - SHOULDER_BUFFER) / arm.getConstants().shoulderLength()) <= 1) {
+            targetPos = Math.max(targetPos, -Math.asin((targetElevatorPos - SHOULDER_BUFFER) / arm.getConstants().shoulderLength()));
+        }
+        return targetPos;
+    }
```

Comment:
The rest of the minimum angle calculation methods return the minimum position. This one the lowest value it returns is the target shoulder angle. This should be changed to return the minimum instead, and usages should be changed to `Math.max(...)` where needed.

This just keeps everything consistent.